### PR TITLE
Added retry if fetching token for SPN auth fails with status code 403

### LINKED ISSUE #10860: ARG task fails to fetch access token for SPN if SPN is newly created
## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: Azure Resource group deployment

## Environment
- Server - Azure Pipelines or TFS on-premises?
    
Both

- Agent - Hosted or Private: 
    
Hosted

## Issue Description

ARG task fails to fetch access token for SPN if SPN is newly created. Token api fails with status code 403. This can happen if newly created SPN has not replicated in AAD.

### Task logs

Failed to check the resource group status. Error: \"Could not fetch access token for Azure. Verify if the Service Principal used is valid and not expired.\"

## Troubleshooting
Checkout how to troubleshoot failures and collect debug logs: https://docs.microsoft.com/en-us/vsts/build-release/actions/troubleshooting

### Error logs

[Insert error from the logs here for a quick overview]

### PR DESCRIPTION
**Pull request by hiyadav**
Fixes #10860 

In case of newly created SPN, getting access token can fail with status code 403. This can happen due to SPN replication delay. 
Hence, we should add retry if failure status code is 403.

### CODE REVIEW

**Comment by sachinma** [comment]
Can u link this with an issue

File: Tasks/Common/azure-arm-rest/azure-arm-common.ts [hunk 1]
```diff
@@ -221,7 +221,7 @@ export class ApplicationTokenCredentials {
 
         let webRequestOptions: webClient.WebRequestOptions = {
             retriableErrorCodes: null,
-            retriableStatusCodes: [400, 408, 409, 500, 502, 503, 504],
+            retriableStatusCodes: [400, 403, 408, 409, 500, 502, 503, 504],
```
Comment (bishal-pdMSFT) [inline Tasks/Common/azure-arm-rest/azure-arm-common.ts]: Looks like the error is while fetching resource group existence and not while getting access token - and hence making this change would not fix the underlying issue.
Let me look the release failures to see if this is really the case.

Comment (bishal-pdMSFT) [reply Tasks/Common/azure-arm-rest/azure-arm-common.ts]: I double checked and the reason for release failure is that the resource group has been deleted by user (I confirmed in ARM). This matches with my earlier investigation some time back, but in the meanwhile I was blind-sided due to another issue in AKS CD and assumed that this failure is same. The status code did add to this.
We can choose to not make this change. Sorry for the trouble.

Comment (hiyadav) [reply Tasks/Common/azure-arm-rest/azure-arm-common.ts]: okay

**Comment by bishal-pdMSFT** [comment]
@hiyadav I am re-opening your PR as we are seeing increased failures in ARG task getting access token for SPN.

**Comment by bishal-pdMSFT** [comment]
@Ajay-MS @sachinma @vincentdass please review

**Comment by bishal-pdMSFT** [comment]
#10860

**Comment by sachinma** [comment]
@bishal-pdMSFT did you verify that the issue was indeed while getting token. As per your last observations - it was while talking to ARM?

**Comment by sachinma** [comment]
> @bishal-pdMSFT did you verify that the issue was indeed while getting token. As per your last observations - it was while talking to ARM?

If we are actually seeing this while fetching token - the change looks okay to me

**Comment by bishal-pdMSFT** [comment]
Yes, the error is during fetching taken. 

There was another similar error which was when talking to ARM - more specifically if user has deleted resource group. last time I was talking about this error which required no fix.

**Comment by bishal-pdMSFT** [comment]
@Ajay-MS / @vincentdass / @sachinma can you please review and approve?

**Comment by bishal-pdMSFT** [comment]
> You would probably need to update the task versions.

Yes, I was not aware that we need to update **all** dependent tasks version now.

**Pull request merged**